woodwork/type_sys/inference_functions.py

```diff
@@ -51,7 +51,7 @@ def double_func(series):
 
 
 def boolean_func(series):
-    if boolean_nullable_func(series) and series.notnull().all():
+    if boolean_nullable_func(series) and not series.isnull().any():
```

Comment:
We'll end up doing `not series.isnull().any()` twice for this and the integer inference function. Would it be worth calculating it once in `infer_logical_type` and passing it in as a boolean? (I guess we'd maybe have to deal with kwargs or something in order to not have to change the rest of the inference functions to take it in as well).

Plus there might be a way to combine it with the `dropped_na = col.dropna()` that will also happen in the inference functions called in `infer_logical_type`.

Reply:
Agreed. Less repeated calls, the better our inference will be.

Reply:
I think this will only get called once? A series can't be both `BooleanNullable` and `IntegerNullable`, based on the current inference functions at least, so either `boolean_nullable_func()` or `integer_nullable_func()` should return False and short circuit the `not series.isnull().any()` call.

Reply:
My explanation above might miss the mark a bit. `boolean_func` will generally only get called if `boolean_nullable_func` returns `True`. This could change is someone modifies the parent-child relationship, making `Boolean` a root-level logical type instead a child of `BooleanNullable`, but that's probably unlikely. Same story for the integer functions.

So since `boolean_nullable_func` and `integer_nullable_func` can't both return `True` for a single series, I think this means that only one of `boolean_func` and `integer_func` will get called, meaning this `not series.isnull().any()` call will only happen once.

I think...

Reply:
Got it! That makes sense to me.

Maybe we just make a note of this somewhere so that if we decide to update inference so if `[1,0,1,1]` could be a valid boolean column we're adding this change

Reply:
Yeah, if we go with this approach and merge in this PR we can update #153 to include some info about this.

Reply:
Updated #153 other issue to mention this.